368 - IDS Swaplist Pt-2

jmacaluso711 opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds ids-swaplist component. Currently only click features are supported. The dragging features will be revisited while working on #487 

**Related github/jira issue (required)**:
related to #368 

**Steps necessary to review your pull request (required)**:
- Pull and run branch
- got to http://localhost:4300/ids-swaplist
- select a list item and click the right arrow. The item should be appended to the next list.
- select a list item and click the left arrow. The item should be appended to the previous list. 
- select multiple items and click the left/right arrow. All selected items should be moved to the next/previous list.
- make sure there is sufficient test coverage.

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

<!-- After submitting your PR, please check back to make sure checks on the PR -->

andriivarhanov commented on src/components/ids-list-view/ids-list-view.js:
```diff
@@ -426,6 +429,19 @@ export default class IdsListView extends Base {
     return this.getAttribute(attributes.SELECTABLE);
   }
 
+  set swappable(value) {
```
please add some tests for the attribute and possibly for some other functionality since IdsListView has low test coverage

jmacaluso711 replied:
Actually going to remove this for now.. as we plan to rework the draggable component.

andriivarhanov commented on src/components/ids-swaplist/README.md:
````diff
@@ -9,24 +9,10 @@
 
 ```html
-<ids-swaplist>
+<ids-swaplist count="2">
````
could you please add some description for `count` to the readme and possibly for the component in general

andriivarhanov commented on src/components/ids-swaplist/ids-swaplist.js:
```diff
@@ -49,6 +46,30 @@ export default class IdsSwapList extends Base {
     ];
   }
 
+  swapToNextList(button) {
```
- Please consider making new methods private since public methods could lead to errors and break the component
![image](https://user-images.githubusercontent.com/5204821/149506663-cbaf6b57-c787-4404-ac59-d2740ad53efb.png)

- Also please add some jsdoc comments to the methods

andriivarhanov commented on src/components/ids-swaplist/ids-swaplist.scss:
```diff
@@ -1,7 +1,112 @@
+  margin-inline-start: auto;
+}
+
+ids-button {
```
please add more mixins like `relative()` `flex()` `p-0()` `absolute()` `bg-slate-60()` etc

andriivarhanov reviewed (commented):
Looks good, some minor comments related to documentation and styling. 

Please add some vertical scroll when the list overflowing with items 
![image](https://user-images.githubusercontent.com/5204821/149508015-85cc9312-69ad-4fb6-a0c7-f9a82041494c.png)

tmcconechy commented on src/mixins/sass/ids-general-mixins.scss:
```diff
@@ -117,3 +117,40 @@ $animation-prefixes: -webkit-, -moz-, '';
 }
+
+// Rounded Corners
+@mixin rounded-tr-md {
```
This will soon be in a PR in the design system FYI...

tmcconechy reviewed (commented):
Looks great. The only comments i have for now are

1) the separator (connected lists) has a boxshadow maybe we can remove that?
<img width="184" alt="Screen Shot 2022-01-14 at 2 55 32 PM" src="https://user-images.githubusercontent.com/814283/149577018-b6fe5e98-31e6-43ec-9a73-fc607dfc25b1.png">
2) The keyboard behavior is "almost" working but a bit off (OK as a TODO for later if you want). I think it should work like as before on https://main-enterprise.demo.design.infor.com/components/swaplist/example-index.html

- tab to list -> next tab into the first list -> next tab to second side button > next tab to second list
- once in a list arrow up and down should work -> then space to select

Jah25 commented:
Hey John, looks good I did notice some issues with using keyboard navigation on the swap list, you're able to focus on list items but not move them from list to list with keyboard controls. I'm not sure if that's TODO functionality or not.

jmacaluso711 commented:
@tmcconechy @Jah25 -- I addressed the issues you found. Seems like the list-view keyboard events could use some work, where you can't select an item with keyboard events. May have to revisit in another ticket.

tmcconechy reviewed (approved):
Ok looks good now. Like you say the keyboard looks good but space doesnt select. Could either add a handler for that or put a note in TODO for now.

jmacaluso711 commented:
@tmcconechy looks like there is a note in the list-view TODO file https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-list-view/TODO.md
